server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -212,6 +212,27 @@ func (e *EventChannel) HandleEvent(ctx context.Context, event *pepb.PublishBuild
 		}
 	}
 
+	// When we get the workspace status event, update the invocation in the DB
+	// so that it can be searched by its commit SHA, user name, etc. even
+	// while the invocation is still in progress.
+	if isWorkspaceStatusEvent(bazelBuildEvent) {
+		db := e.env.GetInvocationDB()
+		// TODO: Make sure the lookup/insert happens in a single transaction.
+		// This isn't an issue currently since all invocation-modifying events
+		// are coming through the PublishBuildEvent stream, but using a transaction
+		// here would make this more future-proof.
+		ti, err := db.LookupInvocation(ctx, iid)
+		if err != nil {
+			return err
+		}
+		invocationProto := TableInvocationToProto(ti)
+		event_parser.FillInvocationFromWorkspaceStatus(bazelBuildEvent.GetWorkspaceStatus(), invocationProto)
```

Comment:
I think we want more than just the workspace status event here, I think we should use something more like FillInvocationFromEvents

The dimensions we need can be from any of:
- The workspace status event (if using a workspace_status script).
- The build metadata event (if using build_metadata flags).
- One of the structured command line events (environment variables).

https://www.buildbuddy.io/docs/guide-metadata